src/main/java/com/google/devtools/build/lib/rules/proto/ProtoInfo.java

```diff
@@ -0,0 +1,64 @@
+
+package com.google.devtools.build.lib.rules.proto;
+
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.concurrent.ThreadSafety.Immutable;
+import com.google.devtools.build.lib.events.Location;
+import com.google.devtools.build.lib.skyframe.serialization.autocodec.AutoCodec;
+import com.google.devtools.build.lib.skyframe.serialization.autocodec.AutoCodec.VisibleForSerialization;
+import com.google.devtools.build.lib.skylarkbuildapi.proto.ProtoInfoApi;
+import com.google.devtools.build.lib.syntax.Environment;
+import com.google.devtools.build.lib.syntax.EvalException;
+import com.google.devtools.build.lib.packages.BuiltinProvider;
+import com.google.devtools.build.lib.packages.NativeInfo;
+
+@Immutable
+@AutoCodec
+public final class ProtoInfo extends NativeInfo implements ProtoInfoApi<Artifact> {
+  public static final String SKYLARK_NAME = "ProtoInfo";
+
+  public static final ProtoInfoProvider PROVIDER = new ProtoInfoProvider();
+
+  public static final ProtoInfo EMPTY = ProtoInfo.Builder.create().build();
```

Comment:
That said, this looks like the right way to create a Java provider. How does this not work?

Reply:
It just never appears in the Skylark context.

There's no error messages that I can see, and I don't see any sort of central registry of Skylark symbols, so I'm at a bit of a loss about why Skylark doesn't see this type.

Reply:
Oh, *that* I know :) Just adding a `PROVIDER` field is not enough, you also need to tell Bazel that this should be available as a symbol. Here is now Java does it:

https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/JavaRules.java#L87

Add a `proto_common` top-level symbol should be pretty straightforward by following that template. Let me know if it still doesn't work!